src/lib/Origin.ts

```diff
@@ -16,7 +16,7 @@ export class Origin {
 	}
 
 	public static fromUrl(url: URL): Origin {
-		return new Origin(url.protocol, url.hostname, +url.port);
+		return new Origin(url.protocol, url.hostname.replace(/(^\[|\]$)/g, ''), +url.port);
```

Comment:
The linter CI seems because of single quotes as it seems:

```suggestion
		return new Origin(url.protocol, url.hostname.replace(/(^\[|\]$)/g, ""), +url.port);
```